File: bindings/python/examples/client/07_get_block_data.py
from dataclasses import asdict
from iota_sdk import Client
from dotenv import load_dotenv
import json
import os

load_dotenv()

node_url = os.environ.get('NODE_URL', 'https://api.testnet.shimmer.network')

# Create a Client instance
client = Client(nodes=[node_url])

# Fetch a block ID from the node
block_ids = client.get_tips()
print(f'Block id: {block_ids[0]}')

# Get the metadata for the block
metadata = client.get_block_metadata(block_ids[0])
print(f'Block metadata: {json.dumps(asdict(metadata), indent=4)}')

# Request the block by its id
block = client.get_block_data(block_ids[0])
print(f'Block: {json.dumps(asdict(block), indent=4)}')

File: bindings/nodejs/examples/client/07-get-block-data.ts
// Copyright 2021-2023 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import { Client, initLogger } from '@iota/sdk';
require('dotenv').config({ path: '.env' });

// Run with command:
// yarn run-example ./client/07-get-block-data.ts

// In this example we will send a block and get the data and metadata for it.
async function run() {
    initLogger();
    for (const envVar of ['NODE_URL']) {
        if (!(envVar in process.env)) {
            throw new Error(`.env ${envVar} is undefined, see .env.example`);
        }
    }

    const client = new Client({
        // Insert your node URL in the .env.
        nodes: [process.env.NODE_URL as string],
    });

    try {
        // Fetch a block ID from the node.
        const blockIds = await client.getTips();
        console.log('Block IDs:', blockIds, '\n');

        // Get the metadata for the block.
        const blockMetadata = await client.getBlockMetadata(blockIds[0]);
        console.log('Block metadata: ', blockMetadata, '\n');

        // Request the block by its id.
        const blockData = await client.getBlock(blockIds[0]);
        console.log('Block data: ', blockData, '\n');
    } catch (error) {
        console.error('Error: ', error);
    }
}

run().then(() => process.exit());
